### ISSUE TITLE
Port changes related to the volume mount ownership into the daemonset helm chart template

### ISSUE THREAD

**Issue by parthiban-turfhive** [GoogleCloudPlatform/secrets-store-csi-driver-provider-gcp#385]
### TL;DR
Helm chart deployment fails with the following error because changes of the PR #370 have not been ported to the Helm Chart [daemonset](https://github.com/GoogleCloudPlatform/secrets-store-csi-driver-provider-gcp/blob/main/charts/secrets-store-csi-driver-provider-gcp/templates/daemonset.yaml#L22) template. Also update the app version in the helm chart (it should be 1.5.0 as of filing this bug).

**Expected behavior**
csi store provider container to be up and running with the following message
`{"ts":1708636001493.3962,"caller":"secrets-store-csi-driver-provider-gcp/main.go:89","msg":"starting secrets-store-csi-driver-provider-gcp/v1.5.0","v":0}
{"ts":1708636001793.2656,"caller":"secrets-store-csi-driver-provider-gcp/main.go:232","msg":"health server listening","v":0,"addr":":8095"}`

**Observed behavior**
The container fails to start with the following log message in the error.
`
{"ts":1708634093791.8955,"caller":"secrets-store-csi-driver-provider-gcp/main.go:89","msg":"starting secrets-store-csi-driver-provider-gcp/v1.5.0","v":0}
{"ts":1708634094091.8564,"caller":"secrets-store-csi-driver-provider-gcp/main.go:197","msg":"unable to listen to unix socket","path":"/etc/kubernetes/secrets-store-csi-providers/gcp.sock","err":"listen unix /etc/kubernetes/secrets-store-csi-providers/gcp.sock: bind: permission denied"}
{"ts":1708634094091.9805,"caller":"secrets-store-csi-driver-provider-gcp/main.go:198","msg":"unable to start","v":0}
`

### Reproduction

1.  Deploy using the latest helm chart as of release 1.5.0
2. The application will fail to start with the error messages as mentioned in the observed behavior section.

**Comment by dargudear-google** [comment]
Fixed in #386
